scrunch/order.py

```diff
@@ -0,0 +1,576 @@
+            # Our update to the Hierarchical Order failed. Better reload.
+            self._hier = self._vars = self._graph = None
+            self._load_graph()
+            raise OrderUpdateError(str(e))
+
+    # Proxy methods for the __root__ Group
+
+    def __str__(self):
+        return str(self.graph)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __iter__(self):
+        return self.graph.itervalues()
+
+    def itervalues(self):
+        return self.graph.itervalues()
+
+    def iterkeys(self):
+        return self.graph.iterkeys()
+
+    def keys(self):
+        return self.graph.keys()
+
+    def values(self):
+        return self.graph.values()
+
+    def items(self):
+        return self.graph.items()
+
+    def __getitem__(self, item):
+        return self.graph[item]
+
+
+class DatasetVariablesOrder(Order):
+    def __init__(self, resource):
+        super(DatasetVariablesOrder, self).__init__(resource)
+        self._vars = None
+        self._revision = None
+
+    def _load_hier(self):
+        self._hier = self.resource.session.get(
+            self.resource.variables.orders.hier).payload
+        return self._hier
+
+    def _load_vars(self):
+        self._vars = self.resource.variables.by('id')
+        return self._vars
+
+    @property
+    def vars(self):
+        if self._vars is None:
+            self._load_vars()
+        return self._vars
+
+    @vars.setter
+    def vars(self, _):
+        raise TypeError('Unsupported assignment operation')
+
+    def get(self):
+        # Returns the synchronized hierarchical order graph.
+        if self._sync:
+            ds_state = self.resource.session.get(
+                self.resource.state.self).payload
+            if self._revision is None:
+                self._revision = ds_state.body.revision
+            elif self._revision != ds_state.body.revision:
+                # There's a new dataset revision. Reload the
+                # hierarchical order.
+                self._revision = ds_state.body.revision
+                self._load_hier()
+                self._load_graph()
+        return self
+
+
+class ProjectDatasetsOrder(Order):
+    def __init__(self, resource):
+        super(ProjectDatasetsOrder, self).__init__(resource)
+        self._datasets = None
+
+    def _load_hier(self):
+        self._hier = self.resource.session.get(
+            self.resource.datasets.order.self).payload
+        return self._hier
+
+    def _load_datasets(self):
+        self._datasets = self.resource.datasets.by('id')
+        return self._datasets
+
+    @property
+    def datasets(self):
+        if self._datasets is None:
+            self._load_datasets()
+        return self._datasets
+
+    @datasets.setter
+    def datasets(self, _):
+        raise TypeError('Unsupported assignment operation')
+
+    def get(self):
+        # Returns the synchronized hierarchical order graph.
+        if self._sync:
```

Comment:
Looks like you don't need this `_sync` method, since there is no need for lazy loading the resources, you already have them on init, the instance initialized by itself is good enough.